### ISSUE TITLE
RUF061: false positive when arguments are mis-formed

### ISSUE THREAD

**Issue by woodruffw** [astral-sh/ruff#28146]
### Summary

MRE: [Playground](https://play.ruff.rs/e940dab7-c3ef-4b3c-a2be-caf7474533ed)

Example:

```python
import pytest

with pytest.raises(ValueError, "oops"):
    pass
```

yields:

```
Use context-manager form of `pytest.raises()` (RUF061) [Ln 3, Col 6]
help: Use `pytest.raises()` as a context-manager
```

In this case, the user error is that `"oops"` should be a kwarg, i.e. `match="oops"`. But it's surprising that this gets flagged as RUF061 i.e. for not being a context manager, when it is 🙂 

### Version

0.16.5 (playground)

**Comment by 1678092075** [comment]
Thanks for the report! I was able to reproduce this with Ruff 0.16.5.
I’m planning to add a focused regression test and keep the fix narrowly scoped: RUF061 would ignore pytest.raises(...) calls that are already used as the context expression of a with statement, while leaving the existing behavior unchanged for other calls.
Does that direction match the expected behavior?
